types/logger.d.ts

```diff
@@ -1,5 +1,10 @@
 import { Payload, Plugin } from "./index";
 
+interface ILogger extends Partial<Console> {
```

Comment:
Can we name this just `Logger` instead `I` prefix?

Also, would it be possible to use `Pick` instead of `Partial`, since we require few more methods for the interface.

```ts
interface Logger extends Pick<Console, 'log' | 'group' | 'groupCollapsed' | 'groupEnd'> {}
```

Reply:
I had to make it `Partial<Pick<Console, 'log' | 'group' | 'groupCollapsed' | 'groupEnd>>` since **group**, **groupCollapsed** and **groupEnd** is not compulsory

Reply:
Thanks! OK, yea I think what you mean by those functions are not compulsory, because it would not cause any failure even those methods are not implemented. Hmmm. Not sure what is the best practice in these cases, but I think it make sense with your implementation.

However, I think we should update the interface for the `log` method to match `Console.log` interface. What do you think? And in this case, I think we can strip `log` from `Pick`.

```ts
interface Logger extends Partial<Pick<Console, 'groupCollapsed' | 'group' | 'groupEnd'>> {
  log(...data: any[]): void;
}
```

Reply:
Instead of `log(...data: any[]): void` I went with 
```typescript
interface Logger extends Partial<Pick<Console, 'groupCollapsed' | 'group' | 'groupEnd'>> {
  log(message: string, colour: string, action: any): void;
  log(message: string): void;
}
```
to be a bit more explicit

Reply:
I just remember if the world of JavaScript the signatures are too important so `log(...data: any[]): void` should work as well

Reply:
Ah cool thanks!